nixos/modules/services/databases/neo4j.nix

```diff
@@ -36,81 +89,414 @@ let
+          };
+
+          clientAuth = mkOption {
+            type = types.enum [ "NONE" "OPTIONAL" "REQUIRE" ];
+            default = "REQUIRE";
+            description = ''
+              The client authentication stance for this policy.
+            '';
+          };
+
+          privateKey = mkOption {
+            type = types.str;
+            default = "private.key";
+            description = ''
+              The name of private PKCS #8 key file for this policy to be found
+              in the <literal>baseDirectory</literal>, or the absolute path to
+              the key file. It is mandatory that a key can be found or generated.
+            '';
+          };
+
+          publicCertificate = mkOption {
+            type = types.str;
+            default = "public.crt";
+            description = ''
+              The name of public X.509 certificate (chain) file in PEM format
+              for this policy to be found in the <literal>baseDirectory</literal>,
+              or the absolute path to the certificate file. It is mandatory
+              that a certificate can be found or generated.
+
+              The public certificate is required to be duplicated to the
+              directory holding trusted certificates as defined by the
+              <literal>trustedDir</literal> option.
+            '';
+          };
+
+          revokedDir = mkOption {
+            type = types.path;
+            default = "${cfg.directories.certificates}/__policyname__/revoked";
+            description = ''
+              Path to directory of CRLs (Certificate Revocation Lists) in
+              PEM format, where "__policyname__" in the default path is
+              replaced by the attribute name of this policy. Must be an
+              absolute path.
+            '';
+          };
+
+          tlsVersions = mkOption {
+            type = types.listOf types.str;
+            default = [ "TLSv1.2" ];
+            description = ''
+              Restrict the TLS protocol versions of this policy to those
+              defined here.
+            '';
+          };
+
+          trustAll = mkOption {
+            type = types.bool;
+            default = false;
+            description = ''
+              Makes this policy trust all remote parties. Enabling this is not
+              recommended and the policy's trusted directory will be ignored.
+              Use of this mode is discouraged. It would offer encryption but
+              no security.
+            '';
+          };
+
+          trustedDir = mkOption {
+            type = types.path;
+            default = "${cfg.directories.certificates}/__policyname__/trusted";
+            description = ''
+              Path to directory of X.509 certificates in PEM format for
+              trusted parties, where "__policyname__" in the default path is
+              replaced by the attribute name of this policy. Must be an
+              absolute path.
+            '';
+          };
+
```

Comment:
To get rid of the `__policyname__` thing, use `type = with types; attrsOf (submodule ({ name, ... }: {` in line 354, and add a section here like this:

```nix
config = {
  trustedDir = mkDefault "${cfg.directories.certificates}/${name}/trusted";
  ...
}
```

And remove the defaults declared in the options.